smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpRpcProtocolGenerator.java

```diff
@@ -420,22 +433,57 @@ private void generateErrorDeserializer(GenerationContext context, StructureShape
         writer.write("");
     }
 
+    private boolean isStreamingUnion(GenerationContext context, OperationShape operation) {
```

Comment:
This method only checks streaming union in the output. It can be renamed to `isStreamingUnionOutput`